File: cpp/1170.cpp
class Solution {
private:
    int f(string& s) {
        int n = s.length();
        int freq[26] = {0};
        for (int i = 0; i < n; i++)
            freq[s[i] - 'a']++;
        for (int i = 0; i < 26; i++)
            if (freq[i] != 0)
                return freq[i];
        return 0;
    }

public:
    vector<int> numSmallerByFrequency(vector<string>& q, vector<string>& w) {
        int n = q.size();
        int freq[12] = {0};
        for (string s : w) {
            int t = f(s);
            while (t >= 0)
                freq[t--]++;
        }
        vector<int> res(n);
        for (int i = 0; i < n; i++)
            res[i] = freq[f(q[i]) + 1];
        return res;
    }
};

File: python3/1170.py
class Solution:
    def numSmallerByFrequency(self, q: List[str], w: List[str]) -> List[int]:
        def f(s):
            freq = [0] * 26
            for i in s:
                freq[ord(i) - 97] += 1
            for j in freq:
                if j:
                    return j
            return 0

        freq = [0] * 12
        for i in w:
            for j in range(0, f(i) + 1):
                freq[j] += 1
        return [freq[f(i) + 1] for i in q]
